dll/win32/shimgvw/shimgvw.c

```diff
@@ -421,32 +430,59 @@ static void pSaveImageAs(HWND hwnd)
     free(codecInfo);
 }
 
+static VOID
+pPrintImage(HWND hwnd)
+{
+    /* FIXME */
+}
+
+static VOID
+EnableToolBarButtons(BOOL bEnable)
+{
+    SendMessage(hToolBar, TB_ENABLEBUTTON, IDC_ZOOMP, bEnable);
+    SendMessage(hToolBar, TB_ENABLEBUTTON, IDC_ZOOMM, bEnable);
+
+    SendMessage(hToolBar, TB_ENABLEBUTTON, IDC_ROT1, bEnable);
+    SendMessage(hToolBar, TB_ENABLEBUTTON, IDC_ROT2, bEnable);
+    SendMessage(hToolBar, TB_ENABLEBUTTON, IDC_SAVE, bEnable);
+    SendMessage(hToolBar, TB_ENABLEBUTTON, IDC_PRINT, bEnable);
+}
+
 static VOID
 pLoadImageFromNode(SHIMGVW_FILENODE *node, HWND hwnd)
 {
     WCHAR szTitleBuf[800];
     WCHAR szResStr[512];
-    WCHAR *c;
+    LPWSTR pchFileTitle;
 
-    if (node)
+    if (image)
     {
-        c = wcsrchr(node->FileName, '\\');
-        if (c)
-        {
-            c++;
-        }
+        GdipDisposeImage(image);
+        image = NULL;
+    }
 
-        LoadStringW(hInstance, IDS_APPTITLE, szResStr, ARRAYSIZE(szResStr));
-        StringCbPrintfW(szTitleBuf, sizeof(szTitleBuf), L"%ls%ls%ls", szResStr, L" - ", c);
-        SetWindowTextW(hwnd, szTitleBuf);
+    if (node == NULL)
+    {
+        EnableToolBarButtons(FALSE);
+        return;
+    }
 
-        if (image)
-        {
-            GdipDisposeImage(image);
-        }
+    pLoadImage(node->FileName);
 
-        pLoadImage(node->FileName);
+    LoadStringW(hInstance, IDS_APPTITLE, szResStr, ARRAYSIZE(szResStr));
+    if (image != NULL)
+    {
+        pchFileTitle = PathFindFileNameW(node->FileName);
+        StringCbPrintfW(szTitleBuf, sizeof(szTitleBuf),
+                        L"%ls%ls%ls", szResStr, L" - ", pchFileTitle);
+        SetWindowTextW(hwnd, szTitleBuf);
+    }
+    else
+    {
+        SetWindowTextW(hwnd, szResStr);
     }
+
+    EnableToolBarButtons(image != NULL);
```

Comment:
Not the same button status than Windows XP for the same situation
![image](https://user-images.githubusercontent.com/24843587/79748533-e18ab980-830d-11ea-80b3-90670af530ce.png)

Reply:
I modified `EnableToolBarButtons`.